tests/rados/test_bluestoretool_workflows.py

```diff
@@ -32,155 +37,336 @@ def run(ceph_cluster, **kw):
+                raise AssertionError(
+                    "'bluefs_dedicated_wal' entry in OSD metadata is not 1"
+                )
+
+            if not (
+                osd_metadata["bluefs_wal_dev_node"]
+                == osd_metadata["bluefs_wal_partition_path"]
+            ):
+                log.error(
+                    "bluefs wal device node and bluefs wal partition path do not match"
+                )
+                raise AssertionError(
+                    "bluefs wal device node and bluefs wal partition path do not match"
+                )
+
+            if not osd_metadata["bluefs_wal_devices"] in empty_devices[0]:
+                log.error(f"bluefs WAL device is not pointing to {empty_devices[0]}")
+                raise AssertionError(
+                    f"bluefs WAL device is not pointing to {empty_devices[0]}"
+                )
+
+            log.info(
+                f"Dedicated WAL device successfully added for OSD {osd_id} on {empty_devices[0]}"
+            )
+
+            # Add a new DB device to existing collocated OSD
+            # ceph-bluestore-tool bluefs-bdev-new-db --path osd path --dev-target new-device
+            log.info(
+                f"\n -----------------------------------------------"
+                f"\n Adding a dedicated DB device for OSD.{osd_id}"
+                f"\n -----------------------------------------------"
+            )
+            db_target = lvm_list.pop()
+            log.info(f"DB will be added on device: {db_target}")
+
+            out = bluestore_obj.add_db_device(
+                osd_id=osd_id, new_device=db_target, db_size=db_size
+            )
+            log.info(out)
+            assert "DB device added" in out
+            osd_metadata = ceph_cluster.get_osd_metadata(
+                osd_id=int(osd_id), client=client
+            )
+            log.debug(f"OSD metadata for osd.{osd_id}: \n {osd_metadata}")
+
+            if not int(osd_metadata["bluefs_dedicated_db"]) == 1:
+                log.error("'bluefs_dedicated_db' entry in OSD metadata is not 1")
+                raise AssertionError(
+                    "'bluefs_dedicated_db' entry in OSD metadata is not 1"
+                )
+
+            if not (
+                osd_metadata["bluefs_db_dev_node"]
+                == osd_metadata["bluefs_db_partition_path"]
+            ):
+                log.error(
+                    "bluefs DB device node and bluefs DB partition path do not match"
+                )
+                raise AssertionError(
+                    "bluefs DB device node and bluefs DB partition path do not match"
+                )
+
+            if not osd_metadata["bluefs_db_devices"] in empty_devices[0]:
+                log.error(f"bluefs DB device is not pointing to {empty_devices[0]}")
+                raise AssertionError(
+                    f"bluefs DB device is not pointing to {empty_devices[0]}"
+                )
+
+            log.info(
```

Comment:
Is this the last step that gets executed currently? I see this in logs :
``` 2024-03-13 19:41:40,023 (cephci.test_bluestoretool_workflows) [INFO] - cephci.forked-repo-cbt-waldb-fix.cephci.ceph.ceph.py:1563 - Running command ceph osd metadata -f json-pretty on 10.0.209.191 timeout 600
2024-03-13 19:41:41,208 (cephci.test_bluestoretool_workflows) [INFO] - cephci.forked-repo-cbt-waldb-fix.cephci.ceph.ceph.py:1597 - Command completed successfully
2024-03-13 19:41:41,209 (cephci.test_bluestoretool_workflows) [INFO] - cephci.forked-repo-cbt-waldb-fix.cephci.tests.rados.test_bluestoretool_workflows.py:186 - Dedicated DB device successfully added for OSD 6 on /dev/vde
2024-03-13 19:41:41,209 (cephci.test_bluestoretool_workflows) [INFO] - cephci.forked-repo-cbt-waldb-fix.cephci.tests.rados.test_bluestoretool_workflows.py:482 - Completed verification of Ceph-BlueStore-Tool commands.
2024-03-13 19:41:41,210 (cephci.test_bluestoretool_workflows) [INFO] - cephci.forked-repo-cbt-waldb-fix.cephci.ceph.ceph.py:1563 - Running command podman --version | awk {'print $3'} on 10.0.211.90 timeout 600
2024-03-13 19:41:41,862 (cephci.test_bluestoretool_workflows) [INFO] - cephci.forked-repo-cbt-waldb-fix.cephci.run.py:1038 - Podman Version 4.6.1
2024-03-13 19:41:41,863 (cephci.test_bluestoretool_workflows) [INFO] - cephci.forked-repo-cbt-waldb-fix.cephci.ceph.ceph.py:1563 - Running command docker --version | awk {'print $3'} on 10.0.211.90 timeout 600
2024-03-13 19:41:42,808 (cephci.test_bluestoretool_workflows) [INFO] - cephci.forked-repo-cbt-waldb-fix.cephci.ceph.ceph.py:1563 - Running command ceph --version | awk '{print $3}' on 10.0.209.191 timeout 600
2024-03-13 19:41:43,484 (cephci.test_bluestoretool_workflows) [INFO] - cephci.forked-repo-cbt-waldb-fix.cephci.run.py:1054 - ceph Version 18.2.1-63.el9cp
2024-03-13 19:41:43,487 (cephci.test_bluestoretool_workflows) [INFO] - cephci.forked-repo-cbt-waldb-fix.cephci.run.py:1013 - ceph_clusters_file rerun/None-M49EUK
2024-03-13 19:41:43,488 (cephci.test_bluestoretool_workflows) [INFO] - cephci.forked-repo-cbt-waldb-fix.cephci.run.py:889 - Test <module 'test_bluestoretool_workflows' from '/home/hakumar/Desktop/CEPHQE/forked-repo-cbt-waldb-fix/cephci/tests/rados/test_bluestoretool_workflows.py'> passed ```

Reply:
This is correct, for non-collocated scenario, we are are currently only next addition of dedicated wal and db